src/dftbp/dftb/periodic.F90

```diff
@@ -116,31 +122,24 @@ subroutine TNeighbourList_init(this, nAtom, nInitNeighbour)
   end subroutine TNeighbourList_init
 
 
+#:if WITH_MPI
   !> Deallocates MPI shared memory if required
   subroutine TNeighbourList_final(this)
 
     !> TNeighbourList instance.
     type(TNeighbourList), intent(inout) :: this
 
     if (this%useMpiWindows_) then
-    #:if WITH_MPI
       if (associated(this%iNeighbourMem_)) then
         call this%iNeighbourWin_%free()
       end if
       if (associated(this%neighDist2Mem_)) then
         call this%neighDist2Win_%free()
       end if
-    #:endif
-    else
-      if (associated(this%iNeighbourMem_)) then
-        deallocate(this%iNeighbourMem_)
-      end if
-      if (associated(this%neighDist2Mem_)) then
-        deallocate(this%neighDist2Mem_)
-      end if
```

Comment:
I think, we need this routine even if MPI windows are not used. `*Mem_` are independent of the window stuff.